### PR TITLE
Add functional spec for optional feature control enhancement

### PR DESCRIPTION
**Pull request by pra-moh**

### CODE REVIEW

File: doc/Optional-Feature-Control.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# SONiC Optional Feature Control Enhancement#
```
Comment (jleveque) [inline doc/Optional-Feature-Control.md]: Delete trailing `#` or add a space before it

File: doc/Optional-Feature-Control.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## Revision ##
+
+| Rev | Date     | Author      | Change Description |
+|:---:|:--------:|:-----------:|--------------------|
+| 0.1 | 10/10/19 | Pradnya Mohite | Initial version    |
+
+## Scope ##
+Add support to enable/disable features in sonic. Features like telemtry agent can be optional and this enhancement will provide way to control that.
```
Comment (jleveque) [inline doc/Optional-Feature-Control.md]:
```suggestion
Add support to enable/disable features in sonic. Features like telemetry agent can be optional and this enhancement will provide way to control that.
```

File: doc/Optional-Feature-Control.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+|:---:|:--------:|:-----------:|--------------------|
+| 0.1 | 10/10/19 | Pradnya Mohite | Initial version    |
+
+## Scope ##
+Add support to enable/disable features in sonic. Features like telemtry agent can be optional and this enhancement will provide way to control that. 
+
+### Implementation Details ###
+* Add feature table in config db.  
+  * Modify sonic-cfggen tool to add table and enable the telemetry feature by default.  
+  * For each feature, key is FEATURE|feature name, status :on/off.
```
Comment (jleveque) [inline doc/Optional-Feature-Control.md]: Change status values from "on/off" to "enabled/disabled"

File: doc/Optional-Feature-Control.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+## Scope ##
+Add support to enable/disable features in sonic. Features like telemtry agent can be optional and this enhancement will provide way to control that. 
+
+### Implementation Details ###
+* Add feature table in config db.  
+  * Modify sonic-cfggen tool to add table and enable the telemetry feature by default.  
+  * For each feature, key is FEATURE|feature name, status :on/off.  
+* Add "config feature enable|disable [feature name]" command line.  
+  * Add support for show and config commands.  
+* Add feature watch in hostcfgd to watch config db entry, and start & enable or stop & disable the service.
```
Comment (jleveque) [inline doc/Optional-Feature-Control.md]:
```suggestion
* Add feature in hostcfgd to listen for Config DB FEATURE table entry changes, and enable & start or stop & disable the respective service as appropriate.
```

File: doc/Optional-Feature-Control.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+Add support to enable/disable features in sonic. Features like telemtry agent can be optional and this enhancement will provide way to control that. 
+
+### Implementation Details ###
+* Add feature table in config db.  
+  * Modify sonic-cfggen tool to add table and enable the telemetry feature by default.  
+  * For each feature, key is FEATURE|feature name, status :on/off.  
+* Add "config feature enable|disable [feature name]" command line.  
+  * Add support for show and config commands.  
+* Add feature watch in hostcfgd to watch config db entry, and start & enable or stop & disable the service.  
+  * When hostcfgd first start, it reads all entries in the FEATURE table and compare with current service status, if there is mismatch, hostcfgd will start/stop the service.
```
Comment (jleveque) [inline doc/Optional-Feature-Control.md]:
```suggestion
  * When hostcfgd first starts, it reads all entries in the FEATURE table and compares with current status of each service. If there is mismatch, hostcfgd will enable & start or stop & disable as appropriate. 
```

File: doc/Optional-Feature-Control.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+### Implementation Details ###
+* Add feature table in config db.  
+  * Modify sonic-cfggen tool to add table and enable the telemetry feature by default.  
+  * For each feature, key is FEATURE|feature name, status :on/off.  
+* Add "config feature enable|disable [feature name]" command line.  
+  * Add support for show and config commands.  
+* Add feature watch in hostcfgd to watch config db entry, and start & enable or stop & disable the service.  
+  * When hostcfgd first start, it reads all entries in the FEATURE table and compare with current service status, if there is mismatch, hostcfgd will start/stop the service.  
+  * hostcfgd also dynamically listen to the FEATURE table and change the service status based on the config db change.
```
Comment (jleveque) [inline doc/Optional-Feature-Control.md]: This line seems redundant with line 18. Remove?

File: doc/Optional-Feature-Control.md (hunk 1 above)
Comment (pra-moh) [reply doc/Optional-Feature-Control.md]: added space

File: doc/Optional-Feature-Control.md (hunk 2 above)
Comment (pra-moh) [reply doc/Optional-Feature-Control.md]: updated

File: doc/Optional-Feature-Control.md (hunk 3 above)
Comment (pra-moh) [reply doc/Optional-Feature-Control.md]: updated

File: doc/Optional-Feature-Control.md (hunk 4 above)
Comment (pra-moh) [reply doc/Optional-Feature-Control.md]: updated

File: doc/Optional-Feature-Control.md (hunk 5 above)
Comment (pra-moh) [reply doc/Optional-Feature-Control.md]: updated

File: doc/Optional-Feature-Control.md (hunk 6 above)
Comment (pra-moh) [reply doc/Optional-Feature-Control.md]: removed

**Pull request merged**